OrbitQt/SourceCodeView.h

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_QT_SOURCE_CODE_VIEW_H_
+#define ORBIT_QT_SOURCE_CODE_VIEW_H_
+
+#include <QObject>
+#include <QVector>
+
+#include "DisassemblyReport.h"
+#include "WebEngine/Dialog.h"
+
+namespace orbit_qt {
+
+/* SourceCodeView is a wrapper around web_engine::Dialog for view highlighted code.
+
+   The class exposes three properties (title, source_code, and language) which can
+   be read by the corresponding web part of SourceCodeView. Use SetSourceCode
+   to change any of these three values. The web part will also be notified and
+   will reload accordingly. "language" is an identifier which needs to be recognized
+   by PrismJS to identify the syntax highlighting language. Currently we support
+   "c", "cpp", and "x86asm". Any other value will lead to no highlighting.
+
+   The first and second constructor argument (profile and web_socket_listen_port)
+   are passed through to QWebEngine::Dialog. Check out this class's documentation
+   for more details.
+
+   Example:
+   QWebEngineProfile profile;
+   SourceCodeView view{&profile, std::nullopt};
+   view.SetSourceCode("example.cpp", "int main() {}", "cpp");
+   view.exec(); // Will open the dialog and block until closed.
+*/
+class SourceCodeView : public QObject {
```

Comment:
should we rename it to CodeView (as it is also used for disassembly)

Reply:
That's an excellent idea! I always was uncomfortable with the name but didn't come up with a better one.

Reply:
Haha, yeah, sometimes the obvious ones might just work :)

Reply:
So I renamed everything to CodeView. Thanks!